.github/workflows/deployment.yml

```diff
@@ -0,0 +1,268 @@
+          echo ACR_USER=$(az acr show -n $(az resource list --resource-type 'Microsoft.ContainerRegistry/registries' --query '[0].name' -o tsv) --query name -o tsv) >> $GITHUB_ENV
+          echo WEBAPP=$(az resource list --resource-type 'Microsoft.Web/sites' --query '[?contains(name, `${{ matrix.webapp }}`)].name' -o tsv) >> $GITHUB_ENV
+
+      - name: ACR 🔐
+        uses: azure/docker-login@v1
+        with:
+          login-server: ${{ env.ACR }}
+          username: ${{ env.ACR_USER }}
+          password: ${{ secrets.ACR_PASSWORD }}
+
+      - name: Directory 📁
+        if: ${{ 'portal-ui' == matrix.webapp }}
+        run: echo "DIRECTORY=portal" >> $GITHUB_ENV
+
+      - name: Directory 📁
+        if: ${{ 'portal-ui' != matrix.webapp }}
+        run: echo "DIRECTORY=${{ matrix.webapp }}" >> $GITHUB_ENV
+
+      - name: Artifacts 🗃️
+        working-directory: ./${{ env.DIRECTORY }}
+        run: |
+          # Build images
+          docker build . \
+          -t ${{ env.ACR }}/${{ matrix.webapp }}:${{ github.sha }} \
+          -t ${{ env.ACR }}/${{ matrix.webapp }}:${{ env.FROM }} \
+          --build-arg GITHUB_SHA=${{ github.sha }}
+
+          # Push images
+          docker push ${{ env.ACR }}/${{ matrix.webapp }}:${{ github.sha }}
+          docker push ${{ env.ACR }}/${{ matrix.webapp }}:${{ env.FROM }}
+
+      - name: Slot 🔀
+        uses: azure/cli@v1.0.9
+        with:
+          inlineScript: |
+            # Create new temporary slot
+            az webapp deployment slot create \
+            --slot ${{ github.sha }} \
+            --name ${{ env.WEBAPP }} \
+            --configuration-source ${{ env.WEBAPP }} \
+            --deployment-container-image-name ${{ env.ACR }}/${{ matrix.webapp }}:${{ env.FROM }} \
+            --docker-registry-server-user ${{ env.ACR_USER }} \
+            --docker-registry-server-password ${{ secrets.ACR_PASSWORD }}
+
+            # Swap slot
+            az webapp deployment slot swap \
+            --slot ${{ github.sha }} \
+            --name ${{ env.WEBAPP }} \
+            --action swap
+
+      - name: Cleanup 🧹
+        if: always()
+        uses: azure/cli@v1.0.9
+        with:
+          inlineScript: |
+            # Delete temporary slot
+            az webapp deployment slot delete \
+            --slot ${{ github.sha }} \
+            --name ${{ env.WEBAPP }}
+
+  functions:
+    name: Functions ⚡
+    needs: setup
+    environment: ${{ needs.setup.outputs.environment }}
+    runs-on: [self-hosted, DTFS, deployment]
+
+    env:
+      ENVIRONMENT: ${{ needs.setup.outputs.environment }}
+
+    strategy:
+      max-parallel: 1
+      # Do not cancel in-progress jobs upon failure
+      fail-fast: false
+      # Single dimension matrix
+      matrix:
+        functions: ["acbs-function", "number-generator-function"]
```

Comment:
Irrelevant for this PR but out of curiosity where is ACBS set up right now? Do dev and staging both point to the same version of TFS API?